subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -0,0 +1,1322 @@
+			mqtt_evt->param.puback.message_id,
+			mqtt_evt->result);
+
+		evt.type = AZURE_IOT_HUB_EVT_PUBACK;
+		evt.data.message_id = mqtt_evt->param.puback.message_id;
+
+		azure_iot_hub_notify_event(&evt);
+		break;
+	case MQTT_EVT_SUBACK:
+		LOG_DBG("MQTT_EVT_SUBACK: id = %d result = %d",
+			mqtt_evt->param.suback.message_id,
+			mqtt_evt->result);
+
+		if (on_suback()) {
+			/* Event DPS-related, and should not be propagated */
+			break;
+		}
+
+		evt.type = AZURE_IOT_HUB_EVT_READY;
+		azure_iot_hub_notify_event(&evt);
+		break;
+	default:
+		break;
+	}
+}
+
+#if IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS)
+static struct mqtt_utf8 *user_name_get(void)
+{
+	ssize_t len;
+
+	len = snprintk(user_name_buf, sizeof(user_name_buf), USER_NAME_TEMPLATE,
+		       dps_hostname_get(), conn_config.device_id);
+	if ((len < 0) || (len > sizeof(user_name_buf))) {
+		LOG_ERR("Failed to create user name");
+		return NULL;
+	}
+	user_name.size = len;
+	return &user_name;
+}
+#endif
+
+#if defined(CONFIG_AZURE_IOT_HUB_STATIC_IPV4)
+static int broker_init(bool dps)
+{
+	struct sockaddr_in *broker4 =
+		((struct sockaddr_in *)&broker);
+
+	inet_pton(AF_INET, CONFIG_AZURE_IOT_HUB_STATIC_IPV4_ADDR,
+		  &broker->sin_addr);
+	broker4->sin_family = AF_INET;
+	broker4->sin_port = htons(CONFIG_AZURE_IOT_HUB_PORT);
+
+	LOG_DBG("IPv4 Address %s",
+		log_strdup(CONFIG_AZURE_IOT_HUB_STATIC_IPV4_ADDR));
+
+	return 0;
+}
+#else
+static int broker_init(bool dps)
+{
+	int err;
+	struct addrinfo *result;
+	struct addrinfo *addr;
+	struct addrinfo hints = {
+		.ai_family = AZURE_IOT_HUB_AF_FAMILY,
+		.ai_socktype = SOCK_STREAM
+	};
+
+	if (dps) {
+#if IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS)
+		err = getaddrinfo(CONFIG_AZURE_IOT_HUB_DPS_HOSTNAME,
+				  NULL, &hints, &result);
+		if (err) {
+			LOG_ERR("getaddrinfo, error %d", err);
+			return -ECHILD;
+		}
+#else
+		LOG_ERR("DPS is not enabled");
+
+		return -ENOTSUP;
+#endif /* CONFIG_AZURE_IOT_HUB_DPS */
+	} else {
+		if (IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS)) {
+			err = getaddrinfo(dps_hostname_get(),
+					  NULL, &hints, &result);
+		} else {
+			err = getaddrinfo(CONFIG_AZURE_IOT_HUB_HOSTNAME,
+					  NULL, &hints, &result);
+		}
+	}
+
+	if (err) {
+		LOG_ERR("getaddrinfo, error %d", err);
+		return -ECHILD;
+	}
+
+	addr = result;
+
+	while (addr != NULL) {
+		if ((addr->ai_addrlen == sizeof(struct sockaddr_in)) &&
+		    (AZURE_IOT_HUB_AF_FAMILY == AF_INET)) {
+			struct sockaddr_in *broker4 =
+				((struct sockaddr_in *)&broker);
+			char ipv4_addr[NET_IPV4_ADDR_LEN];
+
+			broker4->sin_addr.s_addr =
+				((struct sockaddr_in *)addr->ai_addr)
+				->sin_addr.s_addr;
+			broker4->sin_family = AF_INET;
+			broker4->sin_port = htons(CONFIG_AZURE_IOT_HUB_PORT);
+
+			inet_ntop(AF_INET, &broker4->sin_addr.s_addr, ipv4_addr,
+				  sizeof(ipv4_addr));
+			LOG_DBG("IPv4 Address found %s", log_strdup(ipv4_addr));
+			break;
+		} else if ((addr->ai_addrlen == sizeof(struct sockaddr_in6)) &&
+			   (AZURE_IOT_HUB_AF_FAMILY == AF_INET6)) {
```

Comment:
`AZURE_IOT_HUB_AF_FAMILY` is statically assigned to AF_INET, are there plans to make it configurable in the future?

Reply:
Removed IPv6 stuff entirely, it's currently not supported by Azure IoT Hub, so it's just noise in the code